src/NFTDriver.sol

```diff
@@ -370,56 +372,88 @@ contract NFTDriver is ERC721Burnable, DriverTransferUtils, Managed {
-    function name() public pure override returns (string memory) {
+    function name() public view override onlyProxy returns (string memory) {
         return "Drips identity";
     }
 
     /// @inheritdoc IERC721Metadata
-    function symbol() public pure override returns (string memory) {
+    function symbol() public view override onlyProxy returns (string memory) {
         return "DHI";
     }
 
-    /// @inheritdoc ERC721Burnable
-    function burn(uint256 tokenId) public override whenNotPaused {
-        super.burn(tokenId);
+    /// @inheritdoc IERC721Metadata
+    function tokenURI(uint256 tokenId) public view override onlyProxy returns (string memory) {
+        return super.tokenURI(tokenId);
     }
 
     /// @inheritdoc IERC721
-    function approve(address to, uint256 tokenId) public override whenNotPaused {
+    function approve(address to, uint256 tokenId) public override onlyProxy {
         super.approve(to, tokenId);
     }
 
     /// @inheritdoc IERC721
-    function safeTransferFrom(address from, address to, uint256 tokenId)
-        public
-        override
-        whenNotPaused
-    {
-        super.safeTransferFrom(from, to, tokenId);
+    function getApproved(uint256 tokenId) public view override onlyProxy returns (address) {
+        return super.getApproved(tokenId);
     }
 
     /// @inheritdoc IERC721
-    function safeTransferFrom(address from, address to, uint256 tokenId, bytes memory data)
+    function setApprovalForAll(address operator, bool approved) public override onlyProxy {
+        super.setApprovalForAll(operator, approved);
+    }
+
+    /// @inheritdoc IERC721
+    function isApprovedForAll(address owner, address operator)
         public
+        view
         override
-        whenNotPaused
+        onlyProxy
+        returns (bool)
     {
-        super.safeTransferFrom(from, to, tokenId, data);
+        return super.isApprovedForAll(owner, operator);
     }
 
     /// @inheritdoc IERC721
-    function setApprovalForAll(address operator, bool approved) public override whenNotPaused {
-        super.setApprovalForAll(operator, approved);
+    function transferFrom(address from, address to, uint256 tokenId) public override onlyProxy {
+        super.transferFrom(from, to, tokenId);
     }
 
     /// @inheritdoc IERC721
-    function transferFrom(address from, address to, uint256 tokenId)
+    function safeTransferFrom(address from, address to, uint256 tokenId)
         public
         override
-        whenNotPaused
+        onlyProxy
     {
-        super.transferFrom(from, to, tokenId);
+        super.safeTransferFrom(from, to, tokenId);
+    }
+
+    /// @inheritdoc IERC721
+    function safeTransferFrom(address from, address to, uint256 tokenId, bytes memory data)
+        public
+        override
+        onlyProxy
+    {
+        super.safeTransferFrom(from, to, tokenId, data);
+    }
+
+    /// @inheritdoc ERC721Burnable
+    function burn(uint256 tokenId) public override onlyProxy {
```

Comment:
Interesting, that `burn` is not part of any interface and it is not required to consider it in the `supportsInterface`

Reply:
Yeah, I guess that it isn't required by ERC-721 so it isn't covered. I suspect that ERC-165 is considered to be a failed standard, so it's avoided in OZ wherever it isn't explicitly required by other standards. I don't think that I have seen ERC-165 used anywhere yet, as in somebody actually calling `supportsInterface` and checking the result.